sdk/iOS/OoyalaSkinSDK/OOUpNextManager.m

```diff
@@ -82,7 +84,37 @@ - (void)goToNextVideo {
   // if upnext has not been dismissed and there is a next video, play the next video.
   LOG(@"Going to next video based on Up Next");
   if (!self.isDismissed && self.nextVideo) {
-    [self.player setEmbedCode:self.nextVideo[embedCodeKey]];
+    
+#warning: old API. Remove when SDK version > 4.46.0_GA. If you need to use old API, uncomment -playForJustChangedItem in -bridgeCurrentItemChangedNotification: of 'OOSkinPlayerObserver'
+    //[self.player setEmbedCode:self.nextVideo[embedCodeKey]];
+    
+    //new API from OOyalaSDK. Available > 4.46.0_GA
+    [self.player setEmbedCode:self.nextVideo[embedCodeKey] withCallback:^(OOOoyalaError *error) {
+      LOG(@"goToNextVideo setEmbedCode got callback");
+      if (error) {
+        LOG(@"❌ Is setEmbedCode is NOT successfull. Error: %@", error.debugDescription);
+      }
+    }];
+    
+    __weak typeof(self) weakSelf = self;
+    NSString *expectedEmbedCode = weakSelf.nextVideo[embedCodeKey];
+    void (^expectedBlock) (OOVideo *currentItem); //params of OOCurrentItemChangedCallback
+    expectedBlock = ^ (OOVideo *currentItem) {
```

Comment:
`^` should be placed next to `(`, i.e. `^(OO..`

Reply:
Done